src/components/v5/common/Navigation/ColonySwitcher/partials/ColonySwitcherAvatar/ColonySwitcherAvatar.tsx

```diff
@@ -0,0 +1,53 @@
+import clsx from 'clsx';
+import React from 'react';
+
+import { useTablet } from '~hooks';
+import ColonyIcon from '~icons/ColonyIcon.tsx';
+import { usePageLayoutContext } from '~v5/frame/PageLayout/context/PageLayoutContext.ts';
+import Avatar from '~v5/shared/Avatar/index.ts';
+
+import { type ColonySwitcherProps } from '../../types.ts';
+
+const displayName =
+  'v5.common.Navigation.ColonySwitcher.partials.ColonyPickerAvatar';
+
+const ColonySwitcherAvatar = ({
+  colonyContext,
+}: Pick<ColonySwitcherProps, 'colonyContext'>) => {
+  const { showMobileColonyPicker } = usePageLayoutContext();
+
+  const isTablet = useTablet();
+
+  const colonyAvatarSrc = colonyContext?.colony?.metadata?.avatar
+    ? colonyContext.colony.metadata?.thumbnail ??
+      colonyContext.colony.metadata?.avatar
+    : undefined;
```

Comment:
Is this correct? Could maybe break it up so less chaining.

Reply:
This is actually on `master` and I pretty much just ported it over. Maybe I'll deal with it in a separate ticket 👌 
![image](https://github.com/user-attachments/assets/63fceb4b-fe15-4acd-843b-89f3ab6f72c0)